### ISSUE TITLE
Diff Tree View

### ISSUE THREAD

**Issue by johnnyaug** [treeverse/lakeFS#2174]
When viewing a diff in the UI - add a side tree view summarizing the diff. 

Thie tree will be used both in the Uncommitted Changes view, and the Compare view.

It should:
* Display a directory-like structure on the left-hand side.
  * Objects are **not** displayed on the left-hand side - only "directories".  
  * Each level in the tree can be **expanded / collapsed**.
  * Preferably, some directories are already expanded by default, up to a "reasonable" depth. In other words, when I open this view I instantly get an idea of how my diff looks [design required].
* In each directory, the number of added/deleted/conflicted files will appear.
  * It's ok if these numbers are calculated only after clicking/expanding a directory.
* Clicking a directory will open the diff under it on the right-hand side.

Goals:
1. Increase visibility of the diff.
2. Make it easier to navigate the diff.

In the future, the uncommitted changes view tree will also allow selecting only some of the directories for commit (see #2512).

Mock-up to demonstrate the concept:
![image](https://user-images.githubusercontent.com/3149885/135760224-3b437285-0b79-440f-a07c-5557e8e168c2.png)

**Comment by ozkatz** [comment]
@johnnyaug out of the 3 bullet points under _it shoud_, I believe only the second one is not done yet. Should we open a separate issue to track it and close this?

**Comment by itaiad200** [comment]
@johnnyaug Is this still valid?

**Comment by johnnyaug** [comment]
@ozkatz, note that the entire tree should be presented in the tree view - not just one level at a time. So actually none of the points are done.

**Comment by itaiad200** [comment]
We'll first implement the tree structure (left-side only) of the diff/uncommitted view, since that will not require any heavy computation on the server-side. 
Objects will be displayed in the tree, but will not be expandable.
No automatic expansion of directories by default when the tree loads.
